### PR TITLE
TWEEL 44: Create workspace API

### PR DESCRIPTION
**Pull request by jordanhilado**
## Description
- Created create workspace api endpoint at `/api/workspace`
- Included validation for if the user has already created a workspace with the specified name
- Created separate controller and routing for workspaces at `workspaceRoutes.js` + `workspaceController.js`

### Postman + MongoDB testing
- The `[Create Workspace] - /workspace` Postman API call has been added to our team collection. The details of this endpoint can be seen and tested [here.](https://dining-philosophers.postman.co/workspace/Tweello~60172c91-5d9e-4f81-b26e-456899cfb3eb/request/29866910-1739d6ce-01ef-4d8b-b377-88aedfbbb6b4?action=share&creator=29866910&ctx=documentation&active-environment=25430117-600c0575-946c-47bb-8c18-4c526644899a)
image

- Documents will be inserted in `tweello.workspaces`:
image

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -15,6 +16,7 @@ app.use(express.urlencoded({ extended: true }));
 app.use(cookieParser());
 
 app.use("/api/users", userRoutes);
+app.use("/api/workspace", workspaceRoutes);
```
Comment (danielskjo) [inline server/app.js]: Can we change it to /api/workspaces

File: server/app.js [hunk 2]
```diff
@@ -4,6 +4,7 @@ import express from "express";
 import connectDB from "./config/db.js";
 import { errorHandler, notFound } from "./middleware/errorMiddleware.js";
 import userRoutes from "./routes/userRoutes.js";
+import workspaceRoutes from "./routes/workspaceRoutes.js";
 
 dotenv.config();
```
Comment (jordanhilado) [inline server/app.js:9]: test

File: server/app.js [hunk 3]
```diff
@@ -4,6 +4,7 @@ import express from "express";
 import connectDB from "./config/db.js";
 import { errorHandler, notFound } from "./middleware/errorMiddleware.js";
 import userRoutes from "./routes/userRoutes.js";
+import workspaceRoutes from "./routes/workspaceRoutes.js";
```
Comment (jordanhilado) [inline server/app.js:7]: test

**Review by danielskjo — APPROVED** [review]
LGTM!

**Pull request merged**